juju/application.py

```diff
@@ -822,9 +839,15 @@ async def refresh(
     upgrade_charm = refresh
 
     async def local_refresh(
-            self, charm_origin=None, force=False, force_series=False,
-            force_units=False,
-            path=None, resources=None):
+            self,
+            *,
+            charm_origin: _definitions.CharmOrigin,
+            force: bool,
+            force_series: bool,
+            force_units: bool,
+            path: Union[Path, str],
+            resources: Optional[Dict[str, str]],
+        ):
```

Comment:
I like adding type annotations, but is it ok to remove the default values for these arguments? Are we assuming it's only called from `refresh`? I think `local_refresh` is a public method.

Reply:
it is a public method, but charmers are unable to use it and shell out instead:

https://github.com/canonical/oidc-gatekeeper-operator/blob/57cde1fb1214c6e4ba749219f3bff67463b89e98/tests/integration/test_charm.py#L149-L162

Reply:
If no one's using it, and you're intentionally removing the default arguments, then I don't see a problem here. Just checking that they weren't removed in error.